torch/csrc/utils/disable_torch_function.cpp

```diff
@@ -119,8 +119,9 @@ PyObject* THPModule_disable_torch_function(PyObject *self, PyObject *a) {
   py::tuple py_args;
   if (args == nullptr) {
     py_args = py::make_tuple();
-  }
-  else {
+  } else if (PyList_CheckExact(args)) {
+    py_args = py::reinterpret_borrow<py::tuple>(PyList_AsTuple(args));
```

Comment:
reinterpret_borrow here is wrong as PyList_AsTuple returns a new reference; if you only borrow it you will leak the allocated tuple here. https://github.com/pybind/pybind11/issues/1201 may be of interest.

Reply:
ah... I need to use reinterpret_steal here, as a move semantics. Thanks a lot for the cheatsheet, it's way clearer than pybind's doc. :)